README.md

```diff
@@ -44,159 +40,158 @@ In addition, the file name should be of the format `time-report-x.csv`,
-  <td>
-    A
-  </td>
-</tr>
-<tr>
-  <td>
-    14/11/2016
-  </td>
-  <td>
-    5
-  </td>
-  <td>
-    1
-  </td>
-  <td>
-    A
-  </td>
-</tr>
-<tr>
-  <td>
-    20/11/2016
-  </td>
-  <td>
-    3
-  </td>
-  <td>
-    2
-  </td>
-  <td>
-    B
-  </td>
-</tr>
-</table>
-
-should produce the following payroll report:
-
-<table>
-<tr>
-  <th>
-    Employee ID
-  </th>
-  <th>
-    Pay Period
-  </th>
-  <th>
-    Amount Paid
-  </th>
-</tr>
-<tr>
-  <td>
-    1
-  </td>
-  <td>
-    1/11/2016 - 15/11/2016
-  </td>
-  <td>
-    $300.00
-  </td>
-</tr>
-  <td>
-    2
-  </td>
-  <td>
-    16/11/2016 - 30/11/2016
-  </td>
-  <td>
-    $90.00
-  </td>
-</tr>
-</table>
-
-Your application should be easy to set up, and should run on either Linux or
-Mac OS X. It should not require any non open-source software.
-
-There are many ways that this application could be built; we ask that you build
-it in a way that showcases one of your strengths. If you enjoy front-end
-development, do something interesting with the interface. If you like
-object-oriented design, feel free to dive deeper into the domain model of this
-problem. We're happy to tweak the requirements slightly if it helps you show
-off one of your strengths.
+We've agreed to build an API with the following endpoints to serve HTTP requests:
+
+1. An endpoint for uploading a file.
+
+   - This file will conform to the CSV specifications outlined in the previous section.
+   - Upon upload, the timekeeping information within the file must be stored to a database for archival purposes.
+   - If an attempt is made to upload a file with the same report ID as a previously uploaded file, this upload should fail with an error message indicating that this is not allowed.
+
+1. An endpoint for retrieving a payroll report structured in the following way:
+
+   _NOTE:_ It is not the responsibility of the API to return html, as we will delegate the visual layout and redering to the front end. The expectation is that this API will only return JSON data.
+
+   - Return a JSON object `payrollReport`.
+   - `payrollReport` will have a single field, `employeeReports`, containing a list of objects with fields `employeeId`, `payPerdiod`, and `amountPaid`.
+   - The `payPeriod` field is an object containing a date interval that is roughly biweekly. Each month has two pay periods; the _first half_ is from the 1st to the 15th inclusive, and the _second half_ is from the 16th to the end of the month, inclusive. `payPeriod` will have two fields to represent this interval: `startDate` and `endDate`.
+   - Each employee should have a single object in `employeeReports` for each pay period that they have recorded hours worked. The `amountPaid` field should contain the sum of the hours worked in that pay period multiplied by the hourly rate for their job group.
+   - If an employee was not paid in a specific pay period, there should not be an object in `employeeReports` for that employee + pay period combination.
```

Comment:
If the csv file contains a row for 0 hours for employee A, should they be present vs containing no records for employee B and having them be absent? Will we guarantee each employee in the CSV has at least one hour?